.github/workflows/build.yaml

```diff
@@ -15,56 +15,82 @@ permissions:
 
 jobs:
   build:
-    name: build kernel #and system
+    name: build kernel and system
     runs-on: namespace-profile-arm64-8x16-2004-caching
     timeout-minutes: 20
     steps:
-    - uses: actions/checkout@v4
+    - name: Checkout repository
+      uses: actions/checkout@v4
       with:
         lfs: true
 
-    - name: get kernel submodule ref
+    - name: Get kernel submodule ref
       id: kernel-submodule
       run: echo "ref=$(git ls-tree HEAD | awk '$4 == "agnos-kernel-sdm845"' | awk '{print $3}')" >> "$GITHUB_OUTPUT"
 
+    - name: Restore cache output boot.img and *.ko
+      id: cache-output
+      uses: actions/cache/restore@v4
+      with:
+        key: output-${{ steps.kernel-submodule.outputs.ref }}
+        path: |
+          output/boot.img
+          output/*.ko
+
     - name: Checkout agnos-kernel-sdm845
+      if: steps.cache-output.outputs.cache-hit != 'true'
```

Comment:
not sure what you mean. since `output/boot.img` and `output/*.ko` has just been restored from cache for the exact kernel ref, checking out the agnos-kernel-sdm845 submodule is for nothing since it will not be re-built.